python/dolfinx/wrappers/common.cpp

```diff
@@ -152,8 +153,7 @@ void common(nb::module_& m)
           nb::arg("global"));
   // dolfinx::common::Timer
   nb::class_<dolfinx::common::Timer>(m, "Timer", "Timer class")
-      .def(nb::init<>())
-      .def(nb::init<std::string>(), nb::arg("task"))
+      .def(nb::init<std::string>(), nb::arg("task") = nb::none())
```

Comment:
Let's remove the default arg in the wrapper layer. Since we wrap the interface in the Python layer, best to handle the default arg there. Defaults in too many places can lead to bugs.

Reply:
This does not work, faced the same problem within #3322 - I'm not sure what causes this or how it may be fixed, but somehow only when the `nb::none()` is provided it properly picks up on the python `None` to cpp `std::nullopt` conversion.

Reply:
Take a look at https://nanobind.readthedocs.io/en/latest/functions.html#none-arguments.